File: cantinhodoauau-python/scripts/import_csv.py
import csv
import os
import sys
from datetime import datetime
import uuid

# Adiciona o diretório pai ao path para poder importar o app
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app.database import SessionLocal
from app.models import Client

def parse_date(date_str):
    if not date_str:
        return None
    clean = date_str.replace('(', '').replace(')', '').strip()
    if not clean:
        return None
    try:
        # Tenta formato DD/MM/YYYY
        return datetime.strptime(clean, "%d/%m/%Y")
    except ValueError:
        return None

def run_import(csv_path):
    if not os.path.exists(csv_path):
        print(f"Arquivo não encontrado: {csv_path}")
        return

    db = SessionLocal()
    try:
        with open(csv_path, mode='r', encoding='utf-8') as f:
            # Detecta separador
            sample = f.read(1024)
            f.seek(0)
            dialect = csv.Sniffer().sniff(sample)
            reader = csv.DictReader(f, dialect=dialect)
            
            headers = [h.lower() for h in reader.fieldnames]
            print(f"Colunas encontradas: {headers}")

            count = 0
            for row in reader:
                # Normaliza chaves para minúsculo
                row = {k.lower(): v for k, v in row.items()}
                
                dog_name = row.get('dog', '').strip()
                tutor_name = row.get('tutor', '').strip()

                if not dog_name:
                    continue

                # Verifica duplicados
                existing = db.query(Client).filter(
                    Client.name == dog_name,
                    Client.tutor_name == tutor_name
                ).first()
                if existing:
                    continue

                # Peso
                weight_str = row.get('peso', '').replace(',', '.')
                weight = float(weight_str) if weight_str else None

                client = Client(
                    id=str(uuid.uuid4()),
                    name=dog_name,
                    tutor_name=tutor_name,
                    tutor_phone=row.get('telefone', ''),
                    tutor_email=row.get('email', ''),
                    tutor_cpf=row.get('cpf', ''),
                    tutor_address=row.get('endereço', row.get('endereco', '')),
                    breed=row.get('raça', row.get('raca', '')),
                    weight=weight,
                    pet_size=row.get('porte'),
                    gender=row.get('gênero', row.get('genero')),
                    castrated=(row.get('castrado', '').lower() == 'sim'),
                    birth_date=parse_date(row.get('nascimento')),
                    entry_date=parse_date(row.get('entrada')) or datetime.utcnow(),
                    
                    # Vacinas
                    last_gripe=parse_date(row.get('gripe')),
                    last_v10=parse_date(row.get('v10')),
                    last_raiva=parse_date(row.get('raiva')),
                    last_giardia=parse_date(row.get('giardia')),
                    last_antipulgas=parse_date(row.get('antipulgas'))
                )
                
                db.add(client)
                count += 1

            db.commit()
            print(f"Importação concluída: {count} novos clientes inseridos.")

    except Exception as e:
        print(f"Erro durante a importação: {e}")
        db.rollback()
    finally:
        db.close()

if __name__ == "__main__":
    csv_file = r'C:\Users\weder\Downloads\clientes_2026-04-06.csv'
    run_import(csv_file)

File: import_csv.js
import { createClient } from '@supabase/supabase-js';
import * as fs from 'fs';

const supabaseUrl = 'https://iwjskowrusppbogiazeg.supabase.co';
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Iml3anNrb3dydXNwcGJvZ2lhemVnIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzMxODQ4NTQsImV4cCI6MjA4ODc2MDg1NH0.Enlau-M2fp7gA7K15s5bURFB_xfiKMbjK0V7DhGfuOw';
const supabase = createClient(supabaseUrl, supabaseKey);

const parseDate = (str) => {
  if (!str) return null;
  const clean = str.replace(/[()]/g, '').trim();
  if (!clean) return null;
  const parts = clean.split('/');
  if (parts.length === 3) {
    const d = new Date(parseInt(parts[2]), parseInt(parts[1]) - 1, parseInt(parts[0]));
    if (!isNaN(d.getTime())) return d.toISOString();
  }
  return null;
};

const VACCINE_LABELS = {
  gripe: 'Gripe',
  v10: 'V10',
  raiva: 'Raiva',
  giardia: 'Giárdia',
  antipulgas: 'Antipulgas'
};

async function run() {
  const csvPath = 'C:\\Users\\weder\\Downloads\\clientes_2026-04-06.csv';
  if (!fs.existsSync(csvPath)) {
    console.error('File not found:', csvPath);
    return;
  }
  const text = fs.readFileSync(csvPath, 'utf8');
  const lines = text.split('\n').filter(l => l.trim());
  const separator = lines[0].includes(';') ? ';' : ',';
  const headers = lines[0].split(separator).map(h => h.trim().replace(/^["']|["']$/g, '').toLowerCase());
  
  const columns = {
    dog: headers.findIndex(h => h === 'dog'),
    tutor: headers.findIndex(h => h === 'tutor'),
    phone: headers.findIndex(h => h === 'telefone'),
    email: headers.findIndex(h => h === 'email'),
    cpf: headers.findIndex(h => h === 'cpf'),
    address: headers.findIndex(h => h === 'endereço' || h.includes('ender')),
    breed: headers.findIndex(h => h === 'raça' || h.includes('raca')),
    weight: headers.findIndex(h => h.includes('peso')),
    size: headers.findIndex(h => h === 'porte'),
    gender: headers.findIndex(h => h === 'gênero' || h === 'genero'),
    castrated: headers.findIndex(h => h === 'castrado'),
    birth: headers.findIndex(h => h === 'nascimento' || h.includes('nasc')),
    entry: headers.findIndex(h => h === 'entrada')
  };
  
  const vaccineIndexes = {};
  for (const k of Object.keys(VACCINE_LABELS)) {
     vaccineIndexes[k] = headers.findIndex(h => h === VACCINE_LABELS[k].toLowerCase() || h === k.toLowerCase());
  }

  const { data: dbClients } = await supabase.from('clients').select('id, name, tutor_name');
  
  const rowsToInsert = [];
  
  for (let i = 1; i < lines.length; i++) {
     const cols = lines[i].split(separator).map(v => v.trim().replace(/^["']|["']$/g, ''));
     const dogName = cols[columns.dog];
     const tutorName = columns.tutor !== -1 ? cols[columns.tutor] : '';
     
     if (!dogName) continue;
     
     const matched = (dbClients || []).filter(c => c.name.toLowerCase() === dogName.toLowerCase() && (!tutorName || (c.tutor_name || '').toLowerCase() === tutorName.toLowerCase()));
     if (matched.length > 0) continue; // skip duplicates
     
     const vaccines = { gripe: null, v10: null, raiva: null, giardia: null, antipulgas: null };
     for (const k of Object.keys(VACCINE_LABELS)) {
       const idx = vaccineIndexes[k];
       if (idx !== -1) {
         const d = parseDate(cols[idx].replace(/\s*\(.*\)/, ''));
         if (d) vaccines[k] = d;
       }
     }
     
     let weight = null;
     if (columns.weight !== -1 && cols[columns.weight]) {
       weight = parseFloat(cols[columns.weight].replace(',', '.'));
       if (isNaN(weight)) weight = null;
     }

     const row = {
       name: dogName,
       tutor_name: tutorName,
       tutor_phone: columns.phone !== -1 ? cols[columns.phone] : '',
       tutor_email: columns.email !== -1 ? cols[columns.email] : '',
       tutor_cpf: columns.cpf !== -1 ? cols[columns.cpf] : '',
       tutor_address: columns.address !== -1 ? cols[columns.address] : '',
       breed: columns.breed !== -1 ? cols[columns.breed] : '',
       weight: weight,
       pet_size: columns.size !== -1 ? (cols[columns.size] || null) : null,
       gender: columns.gender !== -1 ? (cols[columns.gender] || null) : null,
       castrated: columns.castrated !== -1 ? (cols[columns.castrated].toLowerCase() === 'sim') : false,
       birth_date: columns.birth !== -1 ? parseDate(cols[columns.birth]) : null,
       entry_date: columns.entry !== -1 ? (parseDate(cols[columns.entry]) || new Date().toISOString()) : new Date().toISOString(),
       vaccines: vaccines
     };
     
     rowsToInsert.push(row);
  }
  
  if (rowsToInsert.length > 0) {
    console.log(`Inserting ${rowsToInsert.length} clients...`);
    const { error } = await supabase.from('clients').insert(rowsToInsert);
    if (error) {
      console.error('Failed to import clients:', error);
    } else {
      console.log('Successfully inserted all clients. Dates are strictly preserved.');
    }
  } else {
    console.log('No new clients to insert.');
  }
}

run();
